### DISCUSSION TITLE
Support for IAM Roles Anywhere

### DISCUSSION THREAD

**Discussion by ruslansalikhov** [aws/secrets-store-csi-driver-provider-aws#442] [Q&A]
Hello,

Could you please implement support of [Access Management Roles Anywhere](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/introduction.html)

It will simplify the usage of Secret Manager in non-EKS clusters.

**Comment by goyalya** [comment]
@ruslansalikhov - Are there any issues that you are seeing with deployments to non-EKS Clusters?

**Comment by krraghavanai** [comment]
I have been trying this and the driver keeps trying to use IAM authentication instead of the RolesAnywhere.  Minimally having a working example would help.  It's hard to say at the moment if this feature is supported and doesn't work or it is not yet supported.  Is it expected to work?

**Comment by simonmarty** [comment]
Are you able to run [EKS Anywhere](https://anywhere.eks.amazonaws.com/)? If you're running a non EKS cluster with IAM Roles anywhere, how are you creating the IAM OIDC provider?
